src/OpenLoco/src/Map/MapGenerator/PngImage.cpp

```diff
@@ -0,0 +1,93 @@
+#include "PngImage.h"
+#include "Logging.h"
+
+using namespace OpenLoco::Diagnostics;
+
+namespace OpenLoco::World::MapGenerator
+{
+    PngImage::PngImage(std::vector<png_byte> img, int w, int h, int c)
+    {
+        imageData = img;
+        width = w;
+        height = h;
+        channels = c;
+    }
+
+    PngImage::PngImage(int w, int h, int c)
+    {
+        imageData = std::vector<png_byte>(w * h * c);
+        width = w;
+        height = h;
+        channels = c;
+    }
+
+    void PngImage::getPixel(int x, int y, png_byte& red, png_byte& green, png_byte& blue, png_byte& alpha)
+    {
+        int index = (y * width + x) * channels;
+        red = imageData[index];
+        green = imageData[index + 1];
+        blue = imageData[index + 2];
+        alpha = imageData[index + 3];
+    }
+
+    static void my_png_error_handler(png_structp, png_const_charp error_msg)
+    {
+        throw std::runtime_error(error_msg);
+    }
```

Comment:
Nice! Good to get rid of the setjmp/longjmp stuff. Though perhaps we should adhere to the code style, e.g. lowerCamelCase?